specutils/spectra/spectrum1d.py

```diff
@@ -70,18 +72,31 @@ def __init__(self, flux=None, spectral_axis=None, wcs=None,
                  radial_velocity=None, bin_specification=None, **kwargs):
         # Check for pre-defined entries in the kwargs dictionary.
         unknown_kwargs = set(kwargs).difference(
-            {'data', 'unit', 'uncertainty', 'meta', 'mask', 'copy'})
+            {'data', 'unit', 'uncertainty', 'meta', 'mask', 'copy',
+             'extra_coords'})
 
         if len(unknown_kwargs) > 0:
             raise ValueError("Initializer contains unknown arguments(s): {}."
                              "".format(', '.join(map(str, unknown_kwargs))))
 
-        # If the flux (data) argument is a subclass of nddataref (as it would
+        # If the flux (data) argument is already a Spectrum1D (as it would
         # be for internal arithmetic operations), avoid setup entirely.
-        if isinstance(flux, NDDataRef):
+        if isinstance(flux, Spectrum1D):
             super().__init__(flux)
             return
 
+        # Handle initializing from NDCube objects
+        elif isinstance(flux, NDCube):
+            if flux.unit is None:
+                raise ValueError("Input NDCube missing unit parameter")
+
+            # Change the flux array from bare ndarray to a Quantity
+            q_flux = flux.data*u.Unit(flux.unit)
```

Comment:
I suggest instead using this:
```suggestion
            q_flux = flux.data << u.Unit(flux.unit)
```
that's both the "recommended" way to do this now, and here it has the particular advantage of not copying the data (an importand consideration since cubes can be big). Alternatively, could do
```suggestion
            q_flux = u.Quantity(flux.data, unit=flux.unit, copy=copy)
```
and expose `copy` as an initializer keyword.